programs/vote/src/vote_state/mod.rs

```diff
@@ -3808,4 +3935,89 @@ mod tests {
         assert_eq!(vote_state.node_pubkey, new_node_pubkey);
         assert_eq!(vote_state.block_revenue_collector, new_node_pubkey);
     }
+
+    fn create_bls_pubkey_and_proof_of_possession(
+        vote_account_pubkey: &Pubkey,
+    ) -> (
+        [u8; BLS_PUBLIC_KEY_COMPRESSED_SIZE],
+        [u8; BLS_PROOF_OF_POSSESSION_COMPRESSED_SIZE],
+    ) {
+        let bls_keypair = BLSKeypair::new();
+        let bls_pubkey_compressed: BLSPubkeyCompressed = bls_keypair.public.try_into().unwrap();
+        let mut message: Vec<u8> = b"ALPENGLOW".to_vec();
+        message.extend_from_slice(vote_account_pubkey.as_ref());
+        message.extend_from_slice(&bls_pubkey_compressed.0);
+        let proof_of_possession = bls_keypair.proof_of_possession(Some(&message));
+        let proof_of_possession: BLSProofOfPossession = proof_of_possession.into();
+        let proof_of_possession_compressed: BLSProofOfPossessionCompressed =
+            proof_of_possession.try_into().unwrap();
+        (bls_pubkey_compressed.0, proof_of_possession_compressed.0)
+    }
+
+    #[test]
+    fn test_get_and_update_authorized_voter_v4_with_bls() {
```

Comment:
This is optional, but it would be useful to add a test for a reply attack:
- Generate a valid PoP for VoteAccount A.
- Try to use that exact PoP to initialize VoteAccount B.
- Expected Result: Verification failure (because the message derived inside the program will use B's address, but the PoP signed A's address).